hikari/events/auto_mod_events.py

```diff
@@ -0,0 +1,169 @@
+
+    @property
+    def app(self) -> traits.RESTAware:
+        # <<inherited docstring from BaseEvent>>.
+        return self.rule.app
+
+
+@base_events.requires_intents(intents.Intents.AUTO_MODERATION_CONFIGURATION)
+@attrs_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class AutoModRuleUpdateEvent(AutoModEvent):
+    """Event that's fired when an auto-moderation rule is updated."""
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    rule: auto_mod.AutoModRule = attr.field()
+    """Object of the auto-moderation rule which was updated."""
+
+    @property
+    def app(self) -> traits.RESTAware:
+        # <<inherited docstring from BaseEvent>>.
+        return self.rule.app
+
+
+@base_events.requires_intents(intents.Intents.AUTO_MODERATION_CONFIGURATION)
+@attrs_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class AutoModRuleDeleteEvent(AutoModEvent):
+    """Event that's fired when an auto-moderation rule is deleted."""
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    rule: auto_mod.AutoModRule = attr.field()
+    """Object of the auto-moderation rule which was deleted."""
+
+    @property
+    def app(self) -> traits.RESTAware:
+        # <<inherited docstring from BaseEvent>>.
+        return self.rule.app
+
+
+@base_events.requires_intents(intents.Intents.AUTO_MODERATION_EXECUTION)
+@attrs_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class AutoModActionExecutionEvent(AutoModEvent):
+    """Event that's fired when an auto-mod action is executed."""
+
+    app: traits.RESTAware = attr.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attrs_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    guild_id: snowflakes.Snowflake = attr.field(repr=True)
+    """ID of the guild this action was executed in."""
+
+    action: auto_mod.PartialAutoModAction = attr.field(repr=False)
+    """Object of the action which was executed."""
+
+    rule_id: snowflakes.Snowflake = attr.field()
+    """ID of the rule which was triggered."""
+
+    rule_trigger_type: typing.Union[int, auto_mod.AutoModTriggerType] = attr.field(repr=False)
+    """Type of the rule which was triggered."""
+
+    user_id: snowflakes.Snowflake = attr.field(repr=False)
+    """ID of the user who generated the context which triggered this."""
+
+    channel_id: typing.Optional[snowflakes.Snowflake] = attr.field(repr=False)
+    """ID of the channel the matching context was sent to.
+    This will be `builtins.None` if the message was blocked by auto-moderation
+    of the matched content wasn't in a channel.
+    """
```

Comment:
```suggestion
    channel_id: typing.Optional[snowflakes.Snowflake] = attr.field(repr=False)
    """ID of the channel the matching context was sent to.

    This will be [`None`][] if the message was blocked by auto-moderation
    of the matched content wasn't in a channel.
    """
```